bpf/process/string_maps.h

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright Authors of Cilium */
+
+#ifndef STRING_MAPS_H__
+#define STRING_MAPS_H__
+
+#define STRING_MAPS_OUTER_MAX_ENTRIES 8
+
+/*
+ * To facilitate an arbitrary number of strings that can be matched on, string matching
+ * uses a hash look up. The problem with this is that the key to a hash has to be a fixed
+ * size, so if the maximum string length is 128 bytes, then all stored strings will be
+ * 128 bytes long (padded with 0s) and the string to be looked up also has to be padded
+ * with 0s to 128 bytes. This means that a short string will be hashed as if it is 128
+ * bytes long.
+ *
+ * The BPF hash maps use jhash for key hashing. See include/linux/jhash.h. This requires
+ * approximately 1 CPU cycle per byte, so in the example above, hashing every string,
+ * regardless of length, will take ~128 cycles, which is clearly inefficient. See
+ * https://fosdem.org/2023/schedule/event/bpf_hashing/ for details.
+ *
+ * jhash hashes in 12 byte blocks (3 x u32). For all lengths >12, a number of 12 byte
+ * blocks are hashed, and the remainder is hashed using a combination of single byte
+ * loads/shifts, followed by a final mix. It appears that the most efficient use of
+ * jhash is with lengths equal to 12k + 1, minimising the number of single byte loads/
+ * shifts.
```

Comment:
Based in above rationale, wouldn't be isn't 12k+1 resulting in one single byte load? Wouldn't the most efficient use be 12k where the single byte loads are 0?